docs/advanced_topics/documents/title_generation_on_upload.md

```diff
@@ -32,82 +30,174 @@ See MDN for more information about [custom JavaScript events](https://developer.
 
 ## Code examples
 
+For each example below, create the specified external JavaScript file in your app’s static directory, such as static/js/, and reference it in the wagtail_hooks.py file.
+
+### Document specific methods:
+
 ### Adding the file extension to the start of the title
 
 ```python
 # wagtail_hooks.py
 from django.utils.safestring import mark_safe
-
 from wagtail import hooks
-
+from django.templatetags.static import static
 
 @hooks.register("insert_global_admin_js")
 def get_global_admin_js():
-    return mark_safe(
-    """
-    <script>
-    window.addEventListener('DOMContentLoaded', function () {
-        document.addEventListener('wagtail:documents-upload', function(event) {
-            var extension = (event.detail.filename.match(/\.([^.]*?)(?=\?|#|$)/) || [''])[1];
-            var newTitle = '(' + extension.toUpperCase() + ') ' + (event.detail.data.title || '');
-            event.detail.data.title = newTitle;
-        });
+    static_url = static('js/title_with_extension.js')
+    return mark_safe(f'<script src="{static_url}"></script>')
+```
+
+Save the following code as static/js/title_with_extension.js
+
+```javascript
+// title_with_extension.js
+window.addEventListener('DOMContentLoaded', function () {
+    document.addEventListener('wagtail:documents-upload', function(event) {
+        var extension = (event.detail.filename.match(/\.([^.]*?)(?=\?|#|$)/) || [''])[1];
```

Comment:
let is favoured more nowadays, var works for both older and newer versions so I thought I'd use it since people aren't always upto date. Of course, if I'm allowed to change the code, I'll put let in.

Reply:
Let's use `const`. It's going to work all the way back to IE11 so we're safe. 

https://caniuse.com/?search=const

I'm not sure why we originally used `var` I wrote this but maybe it was just to be 'careful'. IE11 is now deprecated anyway so `const` is a safe option.